### ISSUE TITLE
Application id uri for local host

### ISSUE THREAD

**Issue by amansani15** [OfficeDev/TeamsFx#7388]
**Describe the bug**
I need to get access token and it's failing to get one in localhost due to application id uri. I have state.dev.json and state.local.json, I am adding client id and applicationiduri under fx-resource-aad-aap-for-teams section. 

**state.local.json**
"fx-resource-aad-app-for-teams": {
        "clientId": "clientid",
        "applicationIdUris": "api://localhost/clientid"
}

**state.dev.json**
"fx-resource-aad-app-for-teams": {
        "clientId": "clientid",
        "applicationIdUris": "api://cdndomain/clientid"
 }

**manifest.template.json**
"webApplicationInfo": {
    "id": "{{state.fx-resource-aad-app-for-teams.clientId}}",
    "resource": "{{{state.fx-resource-aad-app-for-teams.applicationIdUris}}}"
  }

In both the settings, the client id using is from the app registration under portal (Application (client) ID). The app is registered only for cloud app and I don't see any app registered for local one. The applicationid uri used in dev setting is the same uri id in **expose an api** section under app registration. I am able to get the token successfully for the cloud app, because the applicationid uri matches in the settings in the template and in the app registration.  

If I change the applicationiduri on the portal to "api://localhost/clientid", then I can able to get the token in local version but will fail to get one in cloud app. 

**May I know what is the uri I need to use for localhost? and why there is no app registered for local one?**

**Expected behavior**
Make the application id uri works for both local and cloud app.

**VS Code Extension Information (please complete the following information):**
OS: Windows 10 Pro
VS Code Version 1.72.0
Teams tool kit version: V4.2.1

Any inputs are highly appreciated, thank you.

**Comment by swatDong** [comment]
@amansani15, how did you get the clientId in **state.local.json** and **state.dev.json**? Were those values automatically generated by Teams Toolkit, or you manually added those?

As designed, one AAD application can only have one application id uri, thus could only be used in one environment. So you may need two AAD applications (two clientIds), one for local and another for dev.

Usually, if you debug/F5 your app in VSCode, Teams Toolkit will generate one clientId in **state.local.json**. And if you provision your app in VSCode, Teams Toolkit will generate another clientId in **state.dev.json**. Those are different clientIds.

Or, if you'd like to use your own clientId, please follow this doc (https://github.com/OfficeDev/TeamsFx/wiki/Using-existing-Azure-AD-app-in-TeamsFx-project) to add your values to **config.xxx.json**.

**Comment by amansani15** [comment]
The clientid's in local and dev json files are automatically generated by toolkit. I added this part **fx-resource-aad-app-for-teams** to the existing files. 

When I provisioned the application, it registered app under AAD only for cloud one, it didn't create one for local debug. It used to create two apps one of local and the other for cloud with the older versions of toolkit. 

Is there a way to create one for local automatically using toolkit? 

Thank you

**Comment by swatDong** [comment]
I tried the latest toolkit (v4.2.1) and the quickest way to create one for local should be:
- Clear the **fx-resource-aad-app-for-teams** section from **state.local.json**
- F5 to debug in VSCode

During the debug process, Teams Toolkit will automatically create one for local and set values to **state.local.json**.
And you may see such logs in output window:
<img width="820" alt="image" src="https://user-images.githubusercontent.com/7642967/210935627-f2785520-583c-444b-a7f5-df6710e576b9.png">

**Comment by amansani15** [comment]
@swatDong Thank you for the reply.

My application is just the tab capability, SSO is not included. So, it will not create any AAD app registration. After seeing your message, I remember I did create app registration for cloud manually. So, now I created one for local app as well and updated the application ID uri for local host in the newly created app registration and used this application id in the code for localhost. I did testing and I am able to get the token in the local app as well. Thank you
